### PR TITLE
Heroku

### PR DESCRIPTION
**Pull request by zingales**
Added some boilerplate code to get pushing to Heroku working. Also modified the port it was listening on, because it seems like Heroku defines the port for you.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -45,4 +45,6 @@ Temporary Items
 .apdisk
 
 
-# End of https://www.gitignore.io/api/go,osx
\ No newline at end of file
+# End of https://www.gitignore.io/api/go,osxCerealNotes
```
Comment (atmiguel) [inline .gitignore]: why is `CerealNotes` at the end of this line?

File: .gitignore [hunk 2]
```diff
@@ -45,4 +45,6 @@ Temporary Items
...
+
+CerealNotes
```
Comment (atmiguel) [inline .gitignore]: Is this popping up in the home directory? And not your bin directory?

File: index.go [hunk 3]
```diff
@@ -4,8 +4,25 @@ import (
     "fmt"
     "log"
     "net/http"
+    "os"
+    "strconv"
 )
 
+// get the current listening address or fail if input is not correct
+func determineListenAddress() (string, error) {
+  portStr := os.Getenv("PORT")
```
Comment (atmiguel) [inline index.go]: let's do 4 spaces instead of 2

File: index.go [hunk 4]
```diff
@@ -4,8 +4,25 @@ import (
...
+  
+  if portStr == "" {
+    return "", fmt.Errorf("$PORT not set")
+  }
+  
+  port, err := strconv.Atoi(portStr)
+  if err == nil {
+    return "", fmt.Errorf("$PORT not a valid integer")
```
Comment (atmiguel) [inline index.go]: We could also verify that the Port value is in the range 1024 – 49151, but seems unnecessary

File: index.go [hunk 5]
```diff
@@ -20,9 +37,18 @@ func rootHandler(
 }
 
 func main() {
+    addr, err := determineListenAddress()
```
Comment (atmiguel) [inline index.go]: why not `port` instead of `addr`?

File: index.go [hunk 6]
```diff
@@ -20,9 +37,18 @@ func rootHandler(
...
+        log.Fatal(err)
+    }
+
     // set handler
     http.HandleFunc("/", rootHandler)
 
     // start server
-    log.Fatal(http.ListenAndServe(":8080", nil))
+    log.Printf("Listening on %s...\n", addr)
+    err := http.ListenAndServe(addr, nil)
```
Comment (atmiguel) [inline index.go]: I think this was better when it was part of the if statement

**Review by atmiguel — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 2 above)
Comment (zingales) [reply .gitignore]: Ya its cause i did go build install of go install. We want to use go install for sure. I'm not sure i understand why its like that, but i don't have a strong feeling about it

File: index.go (hunk 4 above)
Comment (zingales) [reply index.go]: ya honestly i felt a bit overboard even validating that it was an int. Maybe i should remove the code. I just feel weird with this arbitrary value set in the env.

File: index.go (hunk 5 above)
Comment (zingales) [reply index.go]: its an address cause it has a preceding ":" and its a string

File: index.go (hunk 6 above)
Comment (zingales) [reply index.go]: will change to if block as discussed.

File: .gitignore (hunk 2 above)
Comment (atmiguel) [reply .gitignore]: so we planning on removing this?

File: index.go (hunk 4 above)
Comment (atmiguel) [reply index.go]: why does it make you feel weird?

File: index.go (hunk 5 above)
Comment (atmiguel) [reply index.go]: after some googling looks like something like `:80` is referred to as the port, and `80` is the port number.

File: .gitignore (hunk 2 above)
Comment (zingales) [reply .gitignore]: yes

File: index.go (hunk 5 above)
Comment (zingales) [reply index.go]: OhhHhHhhHh will change method and variables to port then

**Review by atmiguel — APPROVED** [review]

**Pull request merged**